openflow13/nx_action.go

```diff
@@ -0,0 +1,956 @@
+	NXAST_NOTE             = 8  // Nicira extended action: note
+	NXAST_SET_TUNNEL_V6    = 9  // Nicira extended action: set_tunnel64
+	NXAST_MULTIPATH        = 10 // Nicira extended action: set_tunnel
+	NXAST_AUTOPATH         = 11 // Nicira extended action: multipath
+	NXAST_BUNDLE           = 12 // Nicira extended action: bundle
+	NXAST_BUNDLE_LOAD      = 13 // Nicira extended action: bundle_load
+	NXAST_RESUBMIT_TABLE   = 14 // Nicira extended action: resubmit(port, table)
+	NXAST_OUTPUT_REG       = 15 // Nicira extended action: output:field
+	NXAST_LEARN            = 16 // Nicira extended action: learn
+	NXAST_EXIT             = 17 // Nicira extended action: exit
+	NXAST_DEC_TTL          = 18 // Nicira extended action: dec_ttl
+	NXAST_FIN_TIMEOUT      = 19 // Nicira extended action: output:field
+	NXAST_CONTROLLER       = 20 // Nicira extended action: fin_timeout
+	NXAST_DEC_TTL_CNT_IDS  = 21 // Nicira extended action: dec_ttl(id1,[id2]...)
+	NXAST_PUSH_MPLS        = 23 // Nicira extended action: push_mpls
+	NXAST_POP_MPLS         = 24 // Nicira extended action: pop_mpls
+	NXAST_SET_MPLS_TTL     = 25 // Nicira extended action: set_mpls_ttl
+	NXAST_DEC_MPLS_TTL     = 26 // Nicira extended action: set_mpls_ttl
+	NXAST_STACK_PUSH       = 27 // Nicira extended action: push:src
+	NXAST_STACK_POP        = 28 // Nicira extended action: pop:dst
+	NXAST_SAMPLE           = 29 // Nicira extended action: sample
+	NXAST_SET_MPLS_LABEL   = 30 // Nicira extended action: set_mpls_label
+	NXAST_SET_MPLS_TC      = 31 // Nicira extended action: set_mpls_tc
+	NXAST_OUTPUT_REG2      = 32 // Nicira extended action: output(port=port,max_len)
+	NXAST_REG_LOAD2        = 33 // Nicira extended action: load
+	NXAST_CNJUNCTION       = 34 // Nicira extended action: conjunction
+	NXAST_CT               = 35 // Nicira extended action: ct
+	NXAST_NAT              = 36 // Nicira extended action: nat, need to be along with ct action
+	NXAST_CONTROLLER2      = 37 // Nicira extended action: controller(userdata=xxx,pause)
+	NXAST_SAMPLE2          = 38 // Nicira extended action: sample, support for exporting egress tunnel
+	NXAST_OUTPUT_TRUNC     = 39 // Nicira extended action: truncate output action
+	NXAST_CT_CLEAR         = 43 // Nicira extended action: ct_clear
+	NXAST_CT_RESUBMIT      = 44 // Nicira extended action: resubmit to table in ct
+	NXAST_RAW_ENCAP        = 46 // Nicira extended action: encap
+	NXAST_RAW_DECAP        = 47 // Nicira extended action: decap
+	NXAST_DEC_NSH_TTL      = 48 // Nicira extended action: dec_nsh_ttl
+)
+
+type NXActionHeader struct {
+	*ActionHeader
+	Vendor  uint32
+	Subtype uint16
+}
+
+func (a *NXActionHeader) Header() *ActionHeader {
+	return a.ActionHeader
+}
+
+func (a *NXActionHeader) NXHeader() *NXActionHeader {
+	return a
+}
+
+func (a *NXActionHeader) Len() (n uint16) {
+	return NxActionHeaderLength
+}
+
+func (a *NXActionHeader) MarshalBinary() (data []byte, err error) {
+	data = make([]byte, a.Len())
+	b := make([]byte, 0)
```

Comment:
that doesn't seem necessary to me (same comment for other locations below as well), do you think it helps with readability?

Reply:
I don't get your point. Do you mean `b` is not needed? As I need to Marshal NxActionHeader firstly, and a byte slice is needed in this operation. If I don't use `b`, I don't know how to copy header bytes into the message bytes.

Reply:
I don't understand why you have:
```
b := make([]byte, 0)
b, err = a.ActionHeader.MarshalBinary()
```
instead of simply:
```
b, err := a.ActionHeader.MarshalBinary()
```
Am I missing something?